addons/account/models/account_bank_statement.py

```diff
@@ -564,12 +316,32 @@ class AccountBankStatementLine(models.Model):
         string='Auto-generated Payments',
         help="Payments generated during the reconciliation of this bank statement lines.")
 
+    # == Technical fields ==
+    internal_index = fields.Char(
+        string='Internal Reference',
+        compute='_compute_internal_index',
+        store=True,
+        index=True,
+        help="Technical field used to store the internal reference of the statement line for fast indexing."
+    )
     # == Display purpose fields ==
-    is_reconciled = fields.Boolean(string='Is Reconciled', store=True,
-        compute='_compute_is_reconciled',
+    is_reconciled = fields.Boolean(string='Is Reconciled',
+        compute='_compute_is_reconciled', store=True,
```

Comment:
If you improve the indentation of the current code, do it also for the string keyword then
```suggestion
    is_reconciled = fields.Boolean(
        string='Is Reconciled',
        compute='_compute_is_reconciled', store=True,
```